machine-learning/src/main.py

```diff
@@ -53,38 +67,39 @@ async def root():
 def ping():
     return "pong"
 
+
 @app.post("/image-classifier/tag-image", status_code=200)
 def image_classification(payload: MlRequestBody):
-    model = _get_model(classification_model, "image-classification")
+    model = get_cached_model(classification_model, "image-classification")
     assetPath = payload.thumbnailPath
     return run_engine(model, assetPath)
 
 
 @app.post("/sentence-transformer/encode-image", status_code=200)
 def clip_encode_image(payload: MlRequestBody):
-    model = _get_model(clip_image_model)
+    model = get_cached_model(clip_image_model, "clip")
     assetPath = payload.thumbnailPath
     return model.encode(Image.open(assetPath)).tolist()
 
 
 @app.post("/sentence-transformer/encode-text", status_code=200)
 def clip_encode_text(payload: ClipRequestBody):
-    model = _get_model(clip_text_model)
+    model = get_cached_model(clip_text_model, "clip")
```

Comment:
I believe you need a different string here as well, correct?

Reply:
It's better if the string is "clip" for both since if the image and text models are the same, we don't need to load two instances. For the full CLIP models, you can call `.encode()` with either text or images with the same model. If using a text-only multilingual model, it will have a different name than the image model so it will only be used for the text endpoint.

Reply:
I was thinking about a situation in that the user used a different model for the text model. For example, the image's model is `clip-ViT-B-32` and the text model is `clip-ViT-B-32-multilingual-v1`, then with the way you are getting the model, wouldn't the two models have the same key? 🤔

Reply:
The key is concatenated between the model name and task so the keys would still be different. e.g. the key for the image model would be `clip-ViT-B-32|clip` and the key for the text model would be `clip-ViT-B-32-multilingual-v1|clip`.

Reply:
Ah I see, sorry I missed that